### ISSUE TITLE
How to combine `cesiumBaseUrl` with vite's base option

### ISSUE THREAD

**Issue by fawick** [CesiumGS/cesium-vite-example#13]
For the sake of this text, consider the target for the application to be `https://example.com/foo`.

To achieve that I set https://vite.dev/config/shared-options.html#base to `/foo`.

Running `vite dev` results in HTTP 404 for everything under `/cesiumStatic` then.

Changing `const cesiumBaseUrl = "cesiumStatic";` to `const cesiumBaseUrl = "foo";` results in an console error like this

```
Uncaught (in promise) SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON
    at JSON.parse (<anonymous>)
    at cesium.js?v=0acf86fb:18796:17
```
Building the project and running `vite preview` also 

(My goal is to host the cesium static files in the same folder as `index.html` or a subfolder thereof).

**Comment by jjspace** [comment]
Good question @fawick.

TL;DR: 
```js
// change
CESIUM_BASE_URL: JSON.stringify(`/${cesiumBaseUrl}`),
// to
CESIUM_BASE_URL: JSON.stringify(`/foo/${cesiumBaseUrl}`),
```

The important thing is to set the global variable `CESIUM_BASE_URL` to the full static path that the cesium library can load assets from excluding the domain. In your case, in addition to the default config, it would be `https://example.com/foo/{cesiumStatic}` so the `CESIUM_BASE_URL` would need to be `/foo/{cesiumStatic}`.
https://github.com/CesiumGS/cesium-vite-example/blob/afebdf5ac122d0969eb323a0d02059caa8032766/vite.config.js#L14

In our vite config we use the `viteStaticCopy` plugin to actually copy the necessary files out of `node_modules` into the built directory
https://github.com/CesiumGS/cesium-vite-example/blob/afebdf5ac122d0969eb323a0d02059caa8032766/vite.config.js#L20-L27

But the way that setting `base` works in vite is that it hosts everything in the `dist` directory at the new base path. This causes a mismatch with where we're copying the static files and where they're being hosted which isn't accounted for by default using the same `cesiumBaseUrl` variable in both places. Just changing the value of `cesiumBaseUrl` like you did would result in the files getting copied into `dist/foo/cesiumStatic` and be hosted at `https://example.com/foo/foo/cesiumStatic` which is not what you wanted.

I would leave the vite static copy alone but by setting the `base` to something other than the default `/` you also need to include that expected path in the global variable.

Hope that makes sense.

**Comment by fawick** [comment]
Ah, yes, perfect, that works. And seems to be (embarassingly) obvious in hindsight.  Thanks, @jjspace
